### DISCUSSION TITLE
How to make clangd recognize user-defined header files?

### DISCUSSION THREAD

**Discussion by ghost** [clangd/clangd#1343] [Q&A] [answered]
I'm using macOS and I know header `bits/stdc++.h` is not a standard header file which clang support. But I want to know how can I make clangd recognize it if I put the header file `bits/stdc++.h` in some directory like `/usr/local/include`. I have put the header file `bits/stdc++.h` in `/usr/local/include` but clangd still show error:
```bash
bits/stdc++.h file not found
```
So what is the correct location I should put this custom header in?

I have searched all related issues but got no answer.

Thanks.

**Comment by HighCommander4** [comment, marked as answer]
`bits/stdc++.h` is an internal header in the libstdc++ standard library implementation.

If clangd doesn't find it automatically, that suggests clangd is using the **libc++** standard library implementation instead, which I believe this is the default on macOS.

Trying to use `bits/stdc++.h` together with libc++ is probably not going to work. Instead, you'd need to get clangd to use libstdc++ as its standard library implementation.

This may just be a matter of adding `--stdlib=libstdc++` to the commands used to compile your files. (This can be done using `CompileFlags: Add:` in the [clangd config file](https://clangd.llvm.org/config.html#add), or by arranging for the commands in `compile_commands.json` to contain it in the first place.)

If `--stdlib=libstdc++` doesn't work, the next option I can think of is to use [--query-driver](https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc) to point clangd to the gcc installation whose libstdc++ you want it to use.

**Reply by ghost** [reply]
Thank you very much.
